.github/workflows/newCoreRecursiveBuild.yaml

```diff
@@ -1,42 +1,39 @@
-name: Core Packages Recursive Build
+name: New Core Files Build Test
 
 on:
   workflow_dispatch:
-    inputs:
-      build_type:
-        type: choice
-        description: Build all or only changed files
-        options:
-          - Only changed files
-          - All files
-      necto_type:
-        type: choice
-        description: Live or Development NECTO build
-        options:
-          - Development
-          - Live
-  pull_request:
-    branches:
-      - main
 
 jobs:
-  recursiveBuild:
+  recursiveCoreBuild:
     runs-on: ubuntu-latest
     permissions:
       contents: write
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Ensure full history is fetched, not just the latest commit
 
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
           python-version: '3.x'
 
+      - name: Add GitHub Actions credentials
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+
+      - name: Fetch Main Branch
+        run: git fetch origin main:main  # Ensure the main branch is fetched
+
       - name: Install Dependencies
         run: |
           python -m pip install --upgrade pip
           pip install -r scripts/requirements/shared.txt
+          pip install -r scripts/requirements/databases.txt
+          pip install colorama
+          pip install enums
           sudo apt-get update
           sudo apt-get install p7zip-full
```

Comment:
There seem to be quite a lot of packages here.

Install them like we do for python modules, through a list.

First, add a txt file with the packages to be installed, with following contents:

+ pkg1
+ pkg2
+ pkg3

Then read the file out as a list.

I assume the following should work:

```sh
sudo apt-get install $(cat PATH_TO_PACKAGE_LIST.txt)
## This one seems like it should work too
for i in $(cat PATH_TO_PACKAGE_LIST.txt); do sudo apt-get install $i; done
```
It will make the file much more readable and follow the python module installation we established already.

Reply:
Created new recursive_build.txt file for all packages that are being installed with apt-get.